backend/src/main/java/com/redhat/cloud/notifications/db/repositories/BehaviorGroupRepository.java

```diff
@@ -76,6 +81,10 @@ public UUID getBundleId(String orgId, UUID behaviorGroupId) {
 
     @Transactional
     BehaviorGroup create(String accountId, String orgId, BehaviorGroup behaviorGroup, boolean isDefaultBehaviorGroup) {
+        if (!this.isAllowedToCreateMoreBehaviorGroups(accountId, orgId)) {
+            throw new BadRequestException("behavior group creation limit reached. Please consider deleting unused behavior groups before creating more.");
+        }
```

Comment:
The `accountId`/`orgId` may be `null` if we are creating a system behavior group. Let's skip the check and always allow the BG creation in that case.

Reply:
All right, done! I haven't written a test because the `testCreateAndUpdateAndDeleteDefaultBehaviorGroup` test already covers the new condition on the check.